src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java

```diff
@@ -171,6 +171,12 @@ public ModelingExerciseResource(ModelingExerciseRepository modelingExerciseRepos
         exerciseService.validateGeneralSettings(modelingExercise);
 
         ModelingExercise modelingExerciseBeforeUpdate = modelingExerciseRepository.findByIdElseThrow(modelingExercise.getId());
+
+        // Forbid changes of course exercise belongs to.
+        if (!modelingExerciseBeforeUpdate.getCourseViaExerciseGroupOrCourseMember().getId().equals(modelingExercise.getCourseViaExerciseGroupOrCourseMember().getId())) {
```

Comment:
please use Objects.equals(id1, id2)
see https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-

Reply:
done, thank you. 👍